### DISCUSSION TITLE
Preset Changes Within Same Track in GRID MODE

### DISCUSSION THREAD

**Discussion by ghost** [SynthstromAudible/DelugeFirmware#1214] [Ideas]
Currently, in GRID MODE, for MIDI tracks, it's possible to send different program changes between each pattern within the same track. This functionality enables dynamic preset changes on an external synthesizer, allowing for a versatile and evolving sound palette during performances.

However, for SYNTH tracks within the Deluge, the preset remains static across patterns in the same track. This limitation restricts the creative potential and flexibility that could be achieved if presets could be dynamically changed.

In my opinion, extending this functionality to SYNTH tracks would be incredibly beneficial. It would be a game-changer for live performances, significantly reducing the need for multiple tracks by enabling dynamic preset shifts within a single track throughout a set.

Is such an enhancement feasible?

**Comment by m-m-adams** [comment]
If you go into the clip and change the preset it'll change only for that clip and not the whole track. It does still result in creating a new track but that's unavoidable - synth tracks are linked based on their name while MIDI tracks are linked on their channel, which is why changing PCs doesn't need a new track

**Comment by ghost** [comment]
@m-m-adams 
Thanks for your reply, 

I was imagining this to have in grid mode each column intended for a type of instrument, lead, bass, drums, etc..
Could we imagine a column in grid mode that doesn't necessarily represent a track, but on which any clip can be freely placed?

**Comment by lobit12** [comment]
when you say track do you mean a vertical column in grid mode?

**Comment by boorch** [comment]
i think what @wiliamailysh means is something like this:
- in GRID mode, each pad can be any synth, regardless of which synth the track plays.
- when a "clip" is triggered, if there's a previous clip playing in the same vertical column, the previous clip stops before the next one plays

which sounds super fun. but i imagine it may disconnect the relation between SONG mode/ARRANGER mode and GRID mode. (moving clips back and forth etc) but maybe there's a clever solution.

i hope this doesn't make it more confusing but, it's a bit like the difference between:
- Octatrack using "parts" _(imagine vertical columns in GRID mode as a "part", and clips as a "pattern" in OT. in Deluge, "part" is the Synth and "pattern" is the Clip)_
- Digitakt not using "parts", essentially treating every "pattern" as its own instance. But only 1 pattern can play at a time. So every "vertical column" in GRID mode is a Digitakt in a way 😅  _(in Deluge terms, any Clip can contain any Synth in the GRID mode, but for each vertical column, every row is a "pattern", so only 1 clip can play on every vertical column, regardless of it hosting the same Synth preset or not)_

**Comment by lobit12** [comment]
I didnt know there was a relationship between song and grid mode.. Does that mean I can do a bunch of clip triggering and have it show up in song mode somehow?

**Reply by boorch** [reply]
the relationship is more related to moving clips from arranger to song view (clips and clip instances), and afaik essentially grid mode is almost like a streamlined and rotated song mode, so the relationship may be in effect there. there may not even be a problem but i'm just cautious/curious 😅

**Reply by lobit12** [reply]
so I just watched a video where it does show that you can record your clip launching performance,  you just have to hit record and then song mode and it seems to do just that.  but you are right in the sense that vertical row 'tracks' are linked so that when you have done your recording it shows them represented horizontally.
